yaml/metadata.go

```diff
@@ -4,15 +4,25 @@
 
 package yaml
 
-import "github.com/go-vela/types/pipeline"
-
-// Metadata is the yaml representation of
-// the metadata block for a pipeline.
-// nolint:lll // jsonschema will cause long lines
-type Metadata struct {
-	Template bool  `yaml:"template,omitempty" json:"template,omitempty" jsonschema:"description=Enables compiling the pipeline as a template.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-template-tag"`
-	Clone    *bool `yaml:"clone,omitempty" json:"clone,omitempty" jsonschema:"default=true,description=Enables injecting the default clone process.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-clone-tag"`
-}
+import (
+	"github.com/go-vela/types/pipeline"
+)
+
+type (
+	// Metadata is the yaml representation of
+	// the metadata block for a pipeline.
+	// nolint:lll // jsonschema will cause long lines
+	Metadata struct {
+		Template    bool     `yaml:"template,omitempty" json:"template,omitempty" jsonschema:"description=Enables compiling the pipeline as a template.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-template-tag"`
+		Clone       *bool    `yaml:"clone,omitempty" json:"clone,omitempty" jsonschema:"default=true,description=Enables injecting the default clone process.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-clone-tag"`
+		Environment []string `yaml:"environment,omitempty" json:"environment,omitempty" jsonschema:"default=true,description=Controls which containers processes can have global env injected.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-environment-tag"`
```

Comment:
this is really an enum, right? can we add that to the `jsonschema` tag? example from elsewhere: https://github.com/go-vela/types/blob/8396842ec2d0693ae1f97f5372debf9577426117/yaml/ruleset.go#L20

we also probably don't want `default=true` - might be copy pasta error

Reply:
sorry, not sure what i was thinking - it's more similar to https://github.com/go-vela/types/blob/84b1e1d2c2bcde72e867d5f0dd945b660db37453/yaml/ruleset.go#L30 and we custom handle that in the schema generator.

at the very least, `default=true` should be removed in the tag.

Reply:
What does the `default=true` mean? Because technically if a user doesn't provide the field we inject

`environment: [ steps, services, secrets ]` into the YAML block. Simplified the compiler code to do that because then we can assume the field is never empty or nil

Reply:
it says the default value is `true`

Reply:
Rmoved https://github.com/go-vela/types/pull/186/commits/76f522399120283e163186af79a40d67d662b1eb